src/math/data2d.cpp

```diff
@@ -355,27 +356,23 @@ bool Data2D::deserialise(LineParser &parser)
     // Read errors / valuse
     if (hasError_)
     {
-        for (auto x = 0; x < x_.size(); ++x)
+        ArrayIndex2D index(values_);
+        while (index++.hasNext())
```

Comment:
Using the `while` loop and our own `hasNext` method is a bit fragile.  It would be better if we could treat this like a standard for loop.  It should look something more like

```suggestion
        for (auto index : ArrayIndex2D(values_))
```

This also offers the coder the option of 

```suggestion
        for (auto &[x, y] : ArrayIndex2D(values_))
```

for when the individual coordinates are more significant (which they are not, in this instance).